tests/Http/Import/ImportRestControllerTest.php

```diff
@@ -275,4 +279,44 @@ public function it_should_throw_an_exception_when_importing_without_id_and_no_js
 
         $this->controller->importWithId($request, '2f66ac1d-fd5a-431e-ba72-821da4176391');
     }
+
+    /**
+     * @test
+     */
+    public function it_reports_an_id_already_in_use(): void
+    {
+        $id = 'c25ea5b8-acd2-4987-a207-6ee11444fde8';
+        $json = json_encode([
+            'name' => [
+                'nl' => 'Voorbeeld naam',
+            ],
+        ], JSON_THROW_ON_ERROR);
+        $request = Request::create('www.uitdatabank.dev', 'GET', [], [], [], [], $json);
+
+        $expectedDocument = new DecodedDocument(
+            $id,
+            [
+                '@id' => 'https://io.uitdatabank.be/mock/c25ea5b8-acd2-4987-a207-6ee11444fde8',
+                'name' => [
+                    'nl' => 'Voorbeeld naam',
+                ],
+            ]
+        );
+
+        $this->expectException(ApiProblem::class);
+        $this->expectExceptionMessage('Resource id already in use');
```

Comment:
You can also use the `CultuurNet\UDB3\Http\ApiProblem\AssertApiProblemTrait` trait here to assert an exact ApiProblem

Reply:
Yes, nice suggestion!